test/chart/LineChart.spec.tsx

```diff
@@ -1515,33 +1603,29 @@ describe('<LineChart /> - Rendering two line charts with syncId', () => {
           <Line type="monotone" dataKey="uv" stroke="#ff7300" />
           <Tooltip />
           <XAxis dataKey="name" />
+          {children}
         </LineChart>
-      );
-      const { container } = render(
-        <div>
-          {chart1}
-          {chart2}
-        </div>,
-      );
+      ),
+    );
 
-      const chartWidth = width - margin.left - margin.right;
-      const dotSpacing = chartWidth / (PageData.length - 1);
+    test('should show tooltips for both charts on MouseEnter and hide on MouseLeave', async () => {
+      const { container, wrapperA, wrapperB } = renderTestCase();
 
       // simulate entering just past Page A of Chart1 to test snapping of the cursor line
       expect(container.querySelectorAll('.recharts-tooltip-cursor')).toHaveLength(0);
 
-      const firstChart = container.querySelector('.recharts-wrapper');
+      const firstChart = wrapperA.querySelector('.recharts-wrapper');
       assertNotNull(firstChart);
 
-      fireEvent.mouseOver(firstChart, { bubbles: true, clientX: margin.left + 0.1 * dotSpacing, clientY: height / 2 });
-      vi.advanceTimersByTime(100);
+      hoverOverFirstLineItem(firstChart);
 
       // There are two tooltips - one for each LineChart as they have the same syncId
       expect(container.querySelectorAll('.recharts-tooltip-cursor')).toHaveLength(2);
 
-      // make sure tooltips display the correct values, synced by data value
-      expect(screen.queryByText('300')).toBeTruthy();
```

Comment:
The previous test was expecting `300` to be somewhere in the document. I have no clue where that came from, it doesn't match what I would expect in the Tooltip from looking at the data and expected index. I suspect that this test used to match something from other elements other than the Tooltip itself? I switched it so that now it only looks at the Tooltip payload and nothing else.

Reply:
Sounds good